static/src/javascripts/test/spec/common/commercial/commercial-feature-policies.spec.js

```diff
@@ -0,0 +1,208 @@
+            const switches = commercialFeaturePolicies.getPolicySwitches().defaultAds;
+            _.forOwn(switches, (featureSwitch)=> {
+                expect(featureSwitch).toBe(true);
+            });
+        });
+    });
+
+    describe('Sensitive content policy', ()=> {
+        it('hides all features on a sensitive page', ()=> {
+            config.page.shouldHideAdverts = true;
+            const switches = commercialFeaturePolicies.getPolicySwitches().sensitiveContent;
+            _.forOwn(switches, (featureSwitch)=> {
+                expect(featureSwitch).toBe(false);
+            });
+        });
+
+        it('hides all features on pages in the childrens\' book site', ()=> {
+            config.page.section = 'childrens-books-site';
+            const switches = commercialFeaturePolicies.getPolicySwitches().sensitiveContent;
+            _.forOwn(switches, (featureSwitch)=> {
+                expect(featureSwitch).toBe(false);
+            });
+        });
+
+        it('applies no changes to a non-sensitive page outside the childrens\' book site', ()=> {
+            config.page.shouldHideAdverts = false;
+            config.page.section = 'news';
+            const switches = commercialFeaturePolicies.getPolicySwitches().sensitiveContent;
+            expect(switches).toBeUndefined();
+        });
+    });
+
+    describe('SSL content policy', ()=> {
+        it('hides all features on a non-admin SSL page', ()=> {
+            config.page.isSSL = true;
+            config.page.section = 'news';
+            const switches = commercialFeaturePolicies.getPolicySwitches().sslContent;
+            _.forOwn(switches, (featureSwitch)=> {
+                expect(featureSwitch).toBe(false);
+            });
+        });
+
+        it('applies no changes to a non-SSL page', ()=> {
+            config.page.isSSL = false;
+            const switches = commercialFeaturePolicies.getPolicySwitches().sslContent;
+            expect(switches).toBeUndefined();
+        });
+
+        it('applies no changes to an admin-section SSL page', ()=> {
+            config.page.isSSL = true;
+            config.page.section = 'admin';
+            const switches = commercialFeaturePolicies.getPolicySwitches().sslContent;
+            expect(switches).toBeUndefined();
+        });
+    });
+
+    describe('UserPrefs policy', ()=> {
+        it('hides all features when userPrefs has "adverts" opt out', ()=> {
+            userPrefs.switchOff('adverts');
+            const switches = commercialFeaturePolicies.getPolicySwitches().userPrefs;
+            _.forOwn(switches, (featureSwitch)=> {
+                expect(featureSwitch).toBe(false);
+            });
+        });
+
+        it('applies no changes otherwise', ()=> {
+            userPrefs.removeSwitch('adverts');
+            const switches = commercialFeaturePolicies.getPolicySwitches().userPrefs;
+            expect(switches).toBeUndefined();
+        });
+    });
+
```

Comment:
The only policy I'm not testing here is the one that implements the 'noads' URI hash, because I can't mock window.location.hash, so I'd need to add in some kind of hash lookup util I could mock instead. It wasn't tested before and I wasn't really sure if it was really worthwhile for something so simple. I don't know what others think.